### DISCUSSION TITLE
No more Calendar entries after Recreating Home-Assistant from a backup

### DISCUSSION THREAD

**Discussion by udochrist** [mampfes/hacs_waste_collection_schedule#1752] [Q&A] [answered]
Hi,

I'm running Home-Assistant-OS on a RPI and have the plugin configured for MyMuell using jumomind.

On Sunday my PI refused to boot due to SD Card issues and I flushed a fresh card and imported the latest backup.
All went well except for the point that the waste_collection integration does no longer show any dates in the calendar.

I've not been able to find any logged statements about error or ok messages regarding the integration.
 Other Calendars (CalDAV based) are working, as is the whole rest of automations and integrations after the restore.

Even triggering the service manually shows no effects in the log or otherwise.

I've already removed and re-added the integration using HACS of course with the appropriate amount of HA restarts. 

Can someone please help me sort this out as it was working perfectly before the restore....

cheers
Udo

**Comment by 5ila5** [comment]
Could you please try our new (tested working) test case (they seemed to change some cities as our old test case) I updated it in #1754

```yaml
waste_collection_schedule:
  sources:
    - name: jumomind_de
      args:
        service_id: mymuell
        city: Bad Wünnenberg-Bleiwäsche
```

**Comment by udochrist** [comment]
Hi,

thanks for picking this up.

Config is not the issue. I've tested both approaches using the id and name based version.
I've even checked out the code and tested by running the jumomind_de.py using the following hack at the end.

```
if __name__ == "__main__":
    # print_md_table()

    s = Source(service_id='mymuell',city='XYZ_PRIVACY_REMOVED_XYZ',street='XYZ_PRIVACY_REMOVED_XYZ')

    print( s.fetch() )
```
All data is retrieved as expected.

I doubt the issue is with retrieving the data. I'm thinking in the direction of whether the integration is called at all.
the only reference in all log files I can find relating to waste_collection_schedule is at the start when ha complains about untested software. I've even put log statements into the code and cannot find anything in the home-assistant.log file

**Comment by udochrist** [comment]
found the log settings. I've added the following to my configuration:

```
logger:
  default: info
  logs:
    custom_components.waste_collection_schedule: debug

```

and restarted:
the only time anything gets logged about waste collection is at startup with:

`2024-02-02 14:57:36.902 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration waste_collection_schedule which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
`
There is no further reference to the waste_collection in all of the logs. even when I trigger the fetch_all manually from the services menu. 
do you have any idea what might be going on or how to remedy the situation?

**Reply by 5ila5** [reply]
```yaml
logger:
  default: info
  logs:
    custom_components.waste_collection_schedule: debug
```
Did not work for me, I needed to use `waste_collection_schedule: debug`. I do not get any debug logs, so I do not think there are any debug log statements in this integration, but you should get them in your logs if you add log statements yourself. (maybe add logger.warning() instead to be sure)


Not running at all would be a pretty odd behavior I haven't seen yet, so I don't really know how to help you there.

**Comment by udochrist** [comment]
Damn.

I think I know what happens: the integration is not adding old dates and it seems that the url for mymuell does not publish current dates

I added a log to see what the call retrieves and lo and behold:

`2024-02-02 20:03:31.255 INFO (SyncWorker_0) [waste_collection_schedule.source.jumomind_de] retrieved entries [Collection{date=2024-01-03, type=Bio Müll}, Collection{date=2024-01-10, type=Restmüll grüner Deckel (4 wö.)}, Collection{date=2024-01-10, type=Restmüll}, Collection{date=2024-01-11, type=Altpapier}, Collection{date=2024-01-11, type=Wertstofftonne}, Collection{date=2024-01-16, type=Bio Müll}, Collection{date=2024-01-23, type=Restmüll}, Collection{date=2024-01-30, type=Bio Müll}]
`

I was not really paying too much attention to the returned data at the first time I called the url.
Dates are all in the past...  I  re-installed my ha instance on 31/01/24. so all dates were past dates all the time....

I think once the current schedule gets published the dates will start to appear again.

**Comment by udochrist** [comment, marked as answer]
And further investigation lead me to the fact that "Aschaffenburg" is as of February no longer publishing the data via the MyMüll app but only their own Stadtwerke-app and their website (where there are ical files available)...

Sorry for all the fuss. But for now I think that Aschaffenburg should be removed from jumomind_de.
